Feature/29 convert `wl_convert_units()` to generic

GegznaV opened the pull request:
Closes #29

GegznaV committed:
Convert wl_convert_units() to generic

GegznaV committed:
Add method wl_convert_units.hyperSpec()

GegznaV committed:
Convert wl_convert_units() to generic

GegznaV committed:
Update documentation

GegznaV committed:
Update list of units

GegznaV committed:
Merge branch 'develop' into feature/29-wl_convert_units-generic

GegznaV committed:
Merge branch 'develop' into feature/29-wl_convert_units-generic

GegznaV committed:
Merge branch 'develop' into feature/29-wl_convert_units-generic

GegznaV committed:
Merge branch 'develop' into feature/29-wl_convert_units-generic

GegznaV committed:
Merge branch 'develop' into feature/29-wl_convert_units-generic

GegznaV commented on R/wl_convert_units.R:
```diff
@@ -35,6 +52,31 @@ wl_convert_units <- function(x, from, to, ref_wl = NULL) {
+  wl(x) <- wl_new
+
+  x@label$.wavelength <-
+    switch(.wl_fix_unit_name(to),
```
Create a separate function that is based on this `switch` sentence.

GegznaV replied:
This will be addressed in a separate PR. See:
- #60

bryanhanson reviewed (commented):
Alright, if I understand this function I see 2 items for your consideration:

1. This line `#' @param x Data for conversion.` would be clearer if it read `#' @param x Data to be converted.  Either a hyperSpec object or a numeric or integer vector.`
2. This UT ` test_that("wl_convert_units.hyperSpec works", {` actually tests both `...units.hyperSpec` and `...units.default`  You may wish to separate them.

I leave it up to you to decide if these items are worth changing.  OK to merge either way.

GegznaV committed:
Update documentation

Addresses: https://github.com/r-hyperspec/hyperSpec/pull/30#pullrequestreview-728792504

GegznaV committed:
Update unit tests

Addresses: https://github.com/r-hyperspec/hyperSpec/pull/30#pullrequestreview-728792504

GegznaV committed:
Merge branch 'develop' into feature/29-wl_convert_units-generic

GegznaV committed:
Merge branch 'develop' into feature/29-wl_convert_units-generic

GegznaV commented:
Thank you, @bryanhanson, for the quick review!
